src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -161,14 +169,18 @@ export const Configurator = ({
         </EuiButton>
         <EuiButton
           onClick={async () => {
-            await queryString.getDatasetService().cacheDataset(dataset);
+            setIsLoading(true);
+            await queryString.getDatasetService().cacheDataset(dataset, services);
+            setIsLoading(false);
             onConfirm(dataset);
           }}
           fill
+          isLoading={isLoading}
+          disabled={isLoading}
         >
           <FormattedMessage
             id="data.explorer.datasetSelector.advancedSelector.confirm"
-            defaultMessage="Select Data"
+            defaultMessage={isLoading ? 'Loading' : 'Select Data'}
```

Comment:
Might need to check with UX on the wording for this, because there would be a loading for the submit button to be ready and then another loading state when we actually run the query.

Reply:
Totally agree, have already changed it once. Would do as a fast follow https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8337#discussion_r1779023524